app.py

```diff
@@ -152,23 +198,37 @@ def parse():
         'server_error': response.get('Error'),
         'version': version(),
         'username': username
-    })
+    }
+
+    if os.getenv ('REDIRECT_AB_TEST') == os.getenv ('HEROKU_APP_NAME') and os.getenv ('HEROKU_APP_NAME'):
+        log_object ['is_test'] = 1
+        if 'x-session_id' in request.headers:
+            log_object ['session'] = request.headers ['x-session_id']
```

Comment:
Doesn't it make more sense to update the global `session` object, rather than just do it here? 

(For example, this logic could have been put into `pre_request`)

Doing it like this makes it so that `log_object['session']` contains something different than `session_id()` returns, which feels like a recipe for confusion later on if we now have 2 different notions of session ids.